src/Database/Redis/ProtocolPipelining.hs

```diff
@@ -93,15 +96,20 @@ connect hostName portID timeoutOpt =
                 Right () -> do
                   phase <- readMVar phaseMVar
                   errConnectTimeout phase
-        hConnect' (PortNumber port) mvar =
+        hConnect' port mvar =
           bracketOnError mkSocket NS.close $ \sock -> do
             NS.setSocketOption sock NS.KeepAlive 1
             void $ swapMVar mvar PhaseResolve
-            host <- BSD.getHostByName hostName
+            hostAddrInfos <- NS.getAddrInfo Nothing (Just hostName) Nothing
+            let getAddr = \ case
+                  [] -> throwIO $ ConnectHostNotFound hostName
```

Comment:
From https://hackage.haskell.org/package/network-2.8.0.0/docs/Network-Socket.html#v:getAddrInfo

> If the query fails, this function throws an IO exception instead of returning an empty list. Otherwise, it returns a non-empty list of AddrInfo values.

Therefore, it's probably safe not to handle the case of an empty list here I guess? (I did the same as you did when I first wrote https://github.com/informatikr/hedis/pull/126 😅)